### PR TITLE
Analytics

### PR DESCRIPTION
**Pull request by mjashanks**
## Description

1. Identifying user, with an annonymous ID (i.e. no longer the api_key)
2. Addition of a few more specifc analytics actions.. e.g. "Table Created", "Added Component", "Automation Block Added" etc

### CODE REVIEW

File: packages/builder/src/components/database/DataTable/popovers/Filter.svelte [hunk 1]
```diff
@@ -63,6 +64,7 @@
     backendUiStore.actions.views.save(view)
     notifier.success(`View ${view.name} saved.`)
     dropdown.hide()
+    analytics.captureEvent("Added View Filter")
```
Comment (shogunpurple) [inline packages/builder/src/components/database/DataTable/popovers/Filter.svelte:67]: NAB: Might be good to add the details of the filter into the event too.

File: packages/builder/src/components/database/DataTable/popovers/Calculate.svelte [hunk 2]
```diff
@@ -35,6 +36,7 @@
   function saveView() {
     backendUiStore.actions.views.save(view)
     notifier.success(`View ${view.name} saved.`)
+    analytics.captureEvent("Added View Calculate")
```
Comment (shogunpurple) [inline packages/builder/src/components/database/DataTable/popovers/Calculate.svelte:39]: NAB: Some details of the calculation might be nice in posthog.

File: packages/builder/src/components/start/CreateAppModal.svelte [hunk 3]
```diff
@@ -14,20 +14,39 @@
...
   export let hasKey
 
+  let isApiKeyValid
+  let lastApiKey
+  let fetchApiKeyPromise
+  const validateApiKey = async apiKey => {
+    if (!apiKey) return false
+
+    // make sure we only fetch once
+    if (isApiKeyValid === undefined || apiKey !== lastApiKey) {
```
Comment (shogunpurple) [inline packages/builder/src/components/start/CreateAppModal.svelte:32]: Doesn't look like `lastApiKey` gets assigned anywhere here. I also don't think you need all this code here. Pretty sure that something like this should do the same thing:

```js
let currentApiKey

const validateApiKey = async key => { 
  if (key === currentApiKey) return
  currentApiKey = key ? await analytics.identifyByApiKey(key) : false;
  return currentApiKey
}

 apiKey: string()
         .required("Please enter your API key.")
         .test("valid-apikey", "This API key is invalid",  validateApiKey),
```

File: packages/builder/src/pages/[application]/deploy/index.svelte [hunk 4]
```diff
@@ -26,10 +26,13 @@
       notifier.success(`Your Deployment is Complete.`)
       deployed = true
       loading = false
-      analytics.captureEvent("web_app_deployment", {
+      analytics.captureEvent("Deployed App", {
         appId,
       })
     } catch (err) {
+      analytics.captureEvent("Deploy App Failed", {
+        appId,
```
Comment (shogunpurple) [inline packages/builder/src/pages/[application]/deploy/index.svelte:34]: An error message could be good here for analytics too.

**Review by shogunpurple — APPROVED** [review]
Nothing wrong - just some suggestions 👍

File: packages/builder/src/pages/[application]/deploy/index.svelte (hunk 4 above)
Comment (mjashanks) [reply packages/builder/src/pages/[application]/deploy/index.svelte:34]: added a poshog capture to all `analytics.captureException` .. so this and others will be logged

File: packages/builder/src/components/start/CreateAppModal.svelte (hunk 3 above)
Comment (mjashanks) [reply packages/builder/src/components/start/CreateAppModal.svelte:32]: Yes - I did try this, and tried again there to be sure.

There is weirdness that happens with svelte reactivity... where `currentApiKey` doesn't get set immediately, even though you've just set it.... i've seen it before too.

But is causes a race condition. awaiting a shared promise is the only way i could get this to work :/

**Pull request merged**